rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -105,97 +110,100 @@ def get_energy_required_to_heat_swh_use(
         if space_id in space_within_building_segment_ids
     ]
 
-    if len(spaces) == 0 and use_units not in REQUIRED_USE_UNIT:
+    if not spaces and use_units not in REQUIRED_USE_UNIT:
         spaces = find_all("$.zones[*].spaces[*]", building_segment)
 
-    energy_required_by_space = {}
-    volume_flow_rate = ZERO.FLOW
     hourly_values = (
         hourly_schedule.get("hourly_values")
         if hourly_schedule is not None
         else [1] * len(inlet_temperature_hourly_values)
     )
+
+    swh_use_value = swh_use.get("use", 0.0)
+    energy_required_by_space = {}
+    volume = ZERO.VOLUME
     for space in spaces:
+        space_id = space["id"]
         if use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_PERSON:
-            power = swh_use_value * space.get("number_of_occupants", 0)
+            power = swh_use_value * ureg("W") * space.get("number_of_occupants", 0)
             energy_required = (
                 power
                 * sum(hourly_values)
+                * ureg("hr")
                 * (1 - drain_heat_recovery_efficiency)
-                * ureg("Btu")
             )
-            energy_required_by_space[space["id"]] = energy_required
+            energy_required_by_space[space_id] = energy_required
 
         elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_AREA:
-            power = swh_use_value * space.get("floor_area", ZERO.AREA).magnitude
+            power = swh_use_value * ureg("W/m2") * space.get("floor_area", ZERO.AREA)
             energy_required = (
                 power
                 * sum(hourly_values)
+                * ureg("hr")
                 * (1 - drain_heat_recovery_efficiency)
-                * ureg("Btu")
             )
-            energy_required_by_space[space["id"]] = energy_required
+            energy_required_by_space[space_id] = energy_required
 
         elif use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER:
-            energy_required_by_space[space["id"]] = (
+            energy_required_by_space[space_id] = (
                 swh_use_value
+                * ureg("W")
                 * sum(hourly_values)
+                * ureg("hr")
                 * (1 - drain_heat_recovery_efficiency)
-                * ureg("Btu")
             )
 
         elif use_units == SERVICE_WATER_HEATING_USE_UNIT.VOLUME_PER_PERSON:
-            volume_flow_rate += swh_use_value * space.get("number_of_occupants", 0)
+            volume += swh_use_value * ureg("m3") * space.get("number_of_occupants", 0)
```

Comment:
SImilar here, should we convert to gallon instead of m3? @yunjoonjung-PNNL

Reply:
The testing case's units are in SI unit. Hence, I believe using ```m3``` is right.

Reply:
In lines 190-203, the calculation of energy I based on the IP unit,  such as WATER_DENSITY = 8.3452 * ureg("lb/gallon")
WATER_SPECIFIC_HEAT = 1.001 * ureg("Btu/lb/delta_degF"). I think these unit should match. Could you double-check?

Reply:
Did you mean the unit should be either IP or SI consistently? Actually, with the ```pint``` package, we don't need to! Please read through the pint package documentation ([link](https://pint.readthedocs.io/en/0.10.1/tutorial.html)).

Reply:
Got it, thanks!